react/features/chat/components/native/MessageContainer.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import md5 from 'js-md5';
+import React, { Component } from 'react';
+import { FlatList } from 'react-native';
+
+import ChatMessage from './ChatMessage';
+import styles from './styles';
+
+type Props = {
+
+    /**
+     * The messages array to render.
+     */
+    messages: Array<Object>
+}
+
+/**
+ * Implements a container to render all the chat messages in a conference.
+ */
+export default class MessageContainer extends Component<Props> {
+    /**
+     * Instantiates a new instance of the component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._renderMessage = this._renderMessage.bind(this);
+    }
+
+    /**
+     * Implements {@code Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <FlatList
+                data = { [ ...this.props.messages ].reverse() }
+                inverted = { true }
+                renderItem = { this._renderMessage }
+                style = { styles.messageContainer } />
+        );
+    }
+
+    _renderMessage: Object => React$Element<*>;
+
+    /**
+     * Renders a single chat message.
+     *
+     * @param {Object} message - The chat message to render.
+     * @returns {React$Element<*>}
+     */
+    _renderMessage({ item: message }) {
+        return (
+            <ChatMessage
+                key = { `key_${md5.hex(JSON.stringify(message))}` }
```

Comment:
doesn't the message have any id of any kind?

Reply:
Unfortunately not... we have a message.id, but that's the sending participant ID (bad design, in my opinion)